src/server/system_services/replication_store.js

```diff
@@ -60,6 +60,16 @@ class ReplicationStore {
         return repl && repl.rules;
     }
 
+    async get_replication_by_prefix_and_destination_bucket(prefix, destination_bucket) {
+        dbg.log1('get_replication_by_prefix_and_destination_bucket: ', prefix, destination_bucket);
+        const query = {
+            "rules.filter.prefix": prefix,
+            "rules.destination_bucket": destination_bucket
+        }
+        const repl = await this._replicationconfigs.findOne( query , { deleted: null });
+        return repl && repl.rules;
+    }
```

Comment:
This will find exact match with prefix.
We want:
1. exact match with the prefix
2. subset of the prefix
3. include in this prefix

Reply:
Done.